dod-ccpo/atat-snow
# Create SNOW subflow ATAT Generate Description of Work and other artifacts

# Overview
This PR and branch represent a cleaned version of the work started in [feature/AT-7344](https://github.com/dod-ccpo/atat-snow/compare/develop...feature/AT-7344), meaning there are no Administrator commits.

## SNOW Artifacts
This PR creates the following ServiceNow artifacts:

- **ATAT Generate Description of Work** (Subflow)
- **ATAT Secure POST Request** (Action)
- **Attach Binary File to Acquisition Package** (Action)

This PR also modifies the **Gather DescriptionOfWork Data** (Action) to modify the schema to match the changes in https://github.com/dod-ccpo/atat-web-api/pull/932

## Testing Process
1. Using your PDI, create a valid acquisition package (fill out all fields).
2. Set up an AWS credential in SNOW (this will require atat-sandbox-admin to retrieve the key value pair from Secrets Manager)
3. Open **ATAT Generate Description of Work** (Subflow) in Flow Designer, and hit Test (in the top right of the screen)
![image](https://user-images.githubusercontent.com/84199040/173138053-4631d0ea-ed98-4298-afe6-aca0ea10e397.png)
4. Open the results, all steps should be completed 
![image](https://user-images.githubusercontent.com/84199040/173140002-6caa8eea-80a3-4524-a966-f86ef0321a8e.png)
5. Check the Acquisition Package record that you specified in the test, it should now have _Description of Work.pdf_ as an attachment 
![image](https://user-images.githubusercontent.com/84199040/173140454-3d2f43af-5383-41ad-9887-d0d6c9af5722.png)
6. Download the attachment, ensure that it opens 
![image](https://user-images.githubusercontent.com/84199040/173140670-79770e27-8392-4fe3-a5e7-4d41d2a26cbf.png)


## Tickets
[AT-7344](https://ccpo.atlassian.net/browse/AT-7344)
[AT-7516](https://ccpo.atlassian.net/browse/AT-7516)
